core/src/main/java/org/elasticsearch/action/admin/cluster/node/tasks/cancel/TransportCancelTasksAction.java

```diff
@@ -111,17 +112,19 @@ protected void processTasks(CancelTasksRequest request, Consumer<CancellableTask
     }
 
     @Override
-    protected synchronized TaskInfo taskOperation(CancelTasksRequest request, CancellableTask cancellableTask) {
+    protected synchronized void taskOperation(CancelTasksRequest request, CancellableTask cancellableTask,
```

Comment:
I switched this to async to support reindex's rethrottle which needs to rethrottle other tasks which, in the future, might be on different nodes. I entertained moving reindex's rethrottle to another base class but it didn't seem worth it as I like being able to rethrottle with all the matching parameters. In particular, I rethrottle the child tasks by setting the parent task on the rethrottle request. Also, I think that other extensions of this subclass might be able to use the async-ness.